src/transformers/models/llama/modeling_llama.py

```diff
@@ -1026,22 +985,21 @@ def forward(
         if inputs_embeds is None:
             inputs_embeds = self.embed_tokens(input_ids)
 
-        if self._use_flash_attention_2:
-            # 2d mask is passed through the layers
-            attention_mask = attention_mask if (attention_mask is not None and 0 in attention_mask) else None
-        elif self._use_sdpa and not output_attentions:
-            # output_attentions=True can not be supported when using SDPA, and we fall back on
-            # the manual implementation that requires a 4D causal mask in all cases.
-            attention_mask = _prepare_4d_causal_attention_mask_for_sdpa(
```

Comment:
By the way, an other reason why it is not good to drop it like that is that in the implementation you have you will never dispatch to flash attention through SDPA since you always pass a custom attention mask, can't we drop the mask for the [relevant cases](https://github.com/huggingface/transformers/blob/ec29d25d9f7109f3fdaadfa51515eb6745a136ba/src/transformers/modeling_attn_mask_utils.py#L371-L381)?

Reply:
Agreed that we'll keep dropping the mask. Think it suffice to check that the first mask of all batch is always 0 -> set to None